cg/meta/demultiplex/demux_post_processing.py

```diff
@@ -184,18 +179,26 @@ def is_flow_cell_directory_valid(self, flow_cell_directory: Path) -> bool:
     def is_demultiplexing_complete(self, flow_cell_directory: Path) -> bool:
         return Path(flow_cell_directory, DemultiplexingDirsAndFiles.DEMUX_COMPLETE).exists()
 
-    def update_sample_read_counts(self, sample_internal_ids: List[str]) -> None:
+    def update_sample_read_counts(self, flow_cell_data: FlowCellDirectoryData) -> None:
         """Update samples in status db with the sum of all read counts for the sample in the sequencing metrics table."""
+
+        q30_threshold: int = FLOWCELL_Q30_THRESHOLD[flow_cell_data.sequencer_type]
+
+        sample_internal_ids: List[str] = self.get_sample_ids_from_sample_sheet(
+            flow_cell_data=flow_cell_data
+        )
+
         for sample_id in sample_internal_ids:
-            self.update_single_sample_read_count(sample_id)
+            self.update_single_sample_read_count(sample_id=sample_id, q30_threshold=q30_threshold)
         self.status_db.session.commit()
 
-    def update_single_sample_read_count(self, sample_id: str) -> None:
+    def update_single_sample_read_count(self, sample_id: str, q30_threshold: int) -> None:
```

Comment:
Add doc str

Reply:
Indeed, I think it makes sense to mention it is fetching data from sample lane sequencing metrics and updating the calculated read counts column in the sample table